packages/ui-stencil/src/components/orama-search-box/orama-search-box.tsx

```diff
@@ -211,47 +259,46 @@ export class SearchBox {
-              focusInput={globalContext.currentTask === 'search'}
-              sourceBaseUrl={this.sourceBaseUrl}
-              disableChat={this.disableChat}
-              suggestions={this.suggestions}
-            />
-            {this.windowWidth <= 1024 && (
-              <orama-chat
-                class={`${globalContext.currentTask === 'chat' ? 'section-active' : 'section-inactive'}`}
-                defaultTerm={globalContext.currentTask === 'chat' ? globalContext.currentTerm : ''}
-                showClearChat={false}
-                focusInput={globalContext.currentTask === 'chat' || chatContext.interactions.length === 0}
-                placeholder={this.placeholder}
-                sourceBaseUrl={this.sourceBaseUrl}
-                sourcesMap={this.sourcesMap}
-                suggestions={this.suggestions}
-              />
-            )}
+            {this.getSearchBox()}
+            {this.windowWidth <= 1024 && this.getChatBox()}
           </div>
           <orama-footer colorScheme={this.colorScheme === 'system' ? this.systemScheme : this.colorScheme} />
         </orama-modal>
-        {this.windowWidth > 1024 && (
-          <orama-sliding-panel
-            open={globalContext.currentTask === 'chat'}
-            closed={() => {
-              globalContext.currentTask = 'search'
-            }}
-          >
-            <orama-chat
-              placeholder={this.placeholder}
-              defaultTerm={globalContext.currentTask === 'chat' ? globalContext.currentTerm : ''}
-              showClearChat={false}
-              sourceBaseUrl={this.sourceBaseUrl}
-              sourcesMap={this.sourcesMap}
-              focusInput={globalContext.currentTask === 'chat' || chatContext.interactions.length === 0}
-              suggestions={this.suggestions}
+        {this.getSlidingPanel()}
+      </Fragment>
+    )
+  }
+
+  getEmbeddedLayout() {
+    return (
+      <Fragment>
+        <div class="embed">
+          {this.disableChat ? null : (
+            <orama-navigation-bar
+              handleClose={this.closeSearchbox}
+              showChatActions={globalContext.currentTask === 'chat'}
             />
-          </orama-sliding-panel>
-        )}
+          )}
+          <div class="main">
+            {this.getSearchBox()}
+            {this.windowWidth <= 1024 && this.getChatBox()}
+          </div>
+          <orama-footer colorScheme={this.colorScheme === 'system' ? this.systemScheme : this.colorScheme} />
+        </div>
+        {this.getSlidingPanel()}
       </Fragment>
     )
   }
+
+  render() {
+    if (!searchState.searchService) {
+      return <orama-text as="p">Unable to initialize search service</orama-text>
+    }
+
+    if (!chatContext.chatService) {
+      return <orama-text as="p">Unable to initialize chat service</orama-text>
+    }
+
+    return this.layout === 'embedded' ? this.getEmbeddedLayout() : this.getModalLayout()
```

Comment:
Probably here I would add only one get function (ie `getInnerContent`) returning this content here:
```
{this.disableChat ? null : (
            <orama-navigation-bar
              handleClose={this.closeSearchbox}
              showChatActions={globalContext.currentTask === 'chat'}
            />
          )}
          <div class="main">
            {this.getSearchBox()}
            {this.windowWidth <= 1024 && this.getChatBox()}
          </div>
          <orama-footer colorScheme={this.colorScheme === 'system' ? this.systemScheme : this.colorScheme} />
```
Since this content is the one that is repeated in both layouts, it would be helpful to move it to a get function. This way, if there are any updates, we only need to review the code in one place.
In this case, we probably don't need getSearchBox, getChatBox, and getSlidingPanel as they simply return a web component, so we would no longer need this extra abstraction.
What do you think?